subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/execution/ProgramParser.kt

```diff
@@ -100,6 +97,40 @@ object ProgramParser {
         }
     }
 
+    private
```

Comment:
🎨 I would move the responsibility to calculate the ranges to erase down to the `Lexer`. This goes with my other comment about `ProgramParserTest` below.

Reply:
I disagree because the `Lexer` should keep its sole responsibility of identifying script pieces. It shouldn't know what we do with those pieces later.